cpp/include/cudf/detail/utilities/alignment.hpp

```diff
@@ -42,5 +42,17 @@ T* align_ptr_for_type(void* destination)
     std::align(alignment, bytes_needed, destination, padded_bytes_needed));
 }
 
+/**
+ * @brief Align a value up to the nearest multiple of the given alignment.
+ *
+ * @param value The input value to align
+ * @param alignment The amount of bytes to align, must be a power of 2
+ * @return The aligned value
+ */
+inline std::size_t align_up(std::size_t value, std::size_t alignment)
+{
```

Comment:
nit:
If `alignment` is going to be compile-time constant, it could be a template parameter and a `static_assert` could be used to enforce it as power of 2 during compilation.
If not, an assert statement to check for power of 2 would be helpful here.